Tomer/memtier set only verify

tomerr-lb opened the pull request:
enable splitting the verify to 2 separate steps, setting the values to verify and getting the values and verify them

tomerr-lb committed:
Memtier_benchmark: added verify-set-only option

This enables us to only perform the set part of the verify benchmark.
this way we can first set all the verify key-values ,
and later on just call the verify only to perform the verification.

ISSUE: LBM1-4057

tomerr-lb committed:
Protocol/memcache_binary: add key not faound response status to errors list.

Added the no key found status to the errors list in the protocol parsing
method.
If we start a get only workload on memtier without first setting the
keys we will get responses from the server containing the key not found
error status.
This status was not marked as an error in the protocol parsing method,
so we would have goten a segfault error when trying to manipulate empty
buffers.

ISSUE: LBM1-4057

tomerr-lb committed:
client: removed unused variable

yael-lb commented on memtier_benchmark.cpp:
```diff
@@ -703,8 +714,8 @@ static int config_parse_args(int argc, char *argv[], struct benchmark_config *cf
-    if (cfg->verify_only && cfg->crc_verify)
-	    cfg->data_verify = 0;
+    if ((cfg->verify_only || cfg->verify_set_only) && cfg->crc_verify)
+        cfg->data_verify = 0;
```
Indentation

sashas-lb reviewed (approved):
LGTM

yael-lb commented on memtier_benchmark.cpp:
```diff
@@ -770,6 +781,7 @@ void usage() {
             "Imported Data Options:\n"
             "      --data-import=FILE         Read object data from file\n"
             "      --data-verify              Enable data verification when test is complete\n"
+            "      --verify-set-only		  Only set the volumes to verify\n"
```
Indentation

yael-lb commented on memtier_benchmark.cpp:
```diff
@@ -104,7 +104,7 @@ static void config_print(FILE *file, struct benchmark_config *cfg)
         "data_import = %s\n"
         "data_verify = %s\n"
-        "verify_only = %s\n"
+        "verify_get_only = %s\n"
```
@tomerr-lb @sashas-lb the verify_only arg is used by the data_verify option of memtier. By changing it's name we break this feature, hence it will make it harder to contribute code if we still plan to. 
Maybe in-order to keep the crc_verify and the data_verify features separated we should add a new argument "crc_verify_get" or something of sorts and restore the use of verfiy_only arg to be used only with data_verify.
אני פשוט חושבת שאנחנו מערבבים שמחה בשמחה וזה יכול ליצור לנו כאב ראש בעתיד...

sashas-lb replied:
mmm @tomerr-lb, @yael-lb  has a good point .. your call guys ..

adir-lb reviewed (commented):
LGTM
please address @yael-lb comments.

tomerr-lb committed:
Memtier_benchmark: enable random key pattern for crc-verify-get/set-only

since we controll the set/get cycles of the crc-verify we should be able to use random key pattern.

ISSUE: LBM1-4074
